freedomofpress/securedrop-client
# add test to ensure db files have restricted 600 perms

# Description

Fixes https://github.com/freedomofpress/securedrop-client/issues/1295

# Test Plan

1. Apply the following diff:
```diff
diff --git a/Makefile b/Makefile
index d59bde7..839958d 100644
--- a/Makefile
+++ b/Makefile
@@ -81,7 +81,7 @@ test: ## Run the application tests in parallel (for rapid development)
 
 .PHONY: test-random
 test-random: ## Run the application tests in random order
-	@TEST_CMD="python -m pytest -v --random-order-bucket=global --ignore=$(FTESTS) --ignore=$(ITESTS) $(TESTOPTS) $(TESTS)" ; \
+	@TEST_CMD="python -m pytest -v --random-order-seed=320201 --random-order-bucket=global --ignore=$(FTESTS) --ignore=$(ITESTS) $(TESTOPTS) $(TESTS)" ; \
 		if command -v xvfb-run > /dev/null; then \
 		xvfb-run -a $$TEST_CMD ; else \
 		$$TEST_CMD ; fi
```
2. Run `make test-random`
3. - [ ] Confirm tests pass with full coverage

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
